src/theme/icon.rs

```diff
@@ -1,13 +1,16 @@
 use serde::Deserialize;
-use std::collections::HashMap;
+use std::collections::{hash_map::IntoIter, HashMap};
+use std::hash::Hash;
 
 #[derive(Debug, Deserialize, PartialEq, Eq)]
 #[serde(rename_all = "kebab-case")]
 #[serde(deny_unknown_fields)]
 #[serde(default)]
 pub struct IconTheme {
-    pub name: HashMap<String, String>,
-    pub extension: HashMap<String, String>,
+    #[serde(deserialize_with = "deserialize_by_name")]
```

Comment:
If we are using a custom deserialize, I think we can avoid creating separate structs.

Reply:
I totally agree with you. I updated the PR. (5e81871)